api/src/iot/import_utils.py

```diff
@@ -30,84 +26,576 @@ def get_center_coordinates(postcode: str) -> Point:
 
 
-CsvRow = namedtuple('CsvRow', ['name', 'categories', 'types', 'x', 'y', 'long', 'lat',
-                               'postalcode', 'housenumber', 'owner_name', 'owner_organisation',
-                               'owner_email', 'contact_name', 'contact_organisation',
-                               'contact_email', ])
+@dataclasses.dataclass
+class PersonData:
+    organisation: str
+    email: str
+    telephone: str
+    website: str
+    first_name: str
+    last_name_affix: str
+    last_name: str
+
+
+@dataclasses.dataclass
+class LatLong:
+    latitude: Union[float, str]
+    longitude: Union[float, str]
+
+
+@dataclasses.dataclass
+class PostcodeHouseNumber:
+    postcode: str
+    house_number: Union[int, str]
+    suffix: str = ''
+
+
+@dataclasses.dataclass
+class LocationDescription:
+    description: str
+
+
+@dataclasses.dataclass
+class Region:
+    region: str
+
+
+@dataclasses.dataclass
+class SensorData:
+    owner: PersonData
+    reference: str
+    type: str
+    location: Union[LatLong, PostcodeHouseNumber, LocationDescription, Region]
+    datastream: str
+    observation_goal: str
+    themes: str
+    contains_pi_data: Literal['Ja', 'Nee']
+    legal_ground: str
+    privacy_declaration: str
+    active_until: Union[datetime.date, str]
+
+
+IPROX_REGISTRATION_FIELDS = [
+    'Verzonden',
+    'Status',
+    'Referentienummer',
+    'Wilt u meer dan 5 sensoren melden?',
+    'Vul uw e-mailadres in',
+]
+
+IPROX_PERSON_FIELDS = [
+    'Naam organisatie/bedrijf',
+    'Postcode',
+    'Huisnummer',
+    'Toevoeging',
+    'Straatnaam',
+    'Plaatsnaam',
+    'E-mail',
+    'Telefoon',
+    'KVK-nummer',
+    'Website',
+    'Voornaam',
+    'Tussenvoegsel',
+    'Achternaam',
+]
+
+IPROX_SENSOR_FIELDS = [
+    'Kies soort / type sensor',
+    'Locatie sensor',
+    'Hebt u een postcode en huisnummer?',
+    'Postcode',
+    'Huisnummer',
+    'Toevoeging',
+    'Omschrijving van de locatie van de sensor',
+    'In welk gebied bevindt zich de mobiele sensor?',
+    'Wat meet de sensor?',
+    'Waarvoor meet u dat?',
+    'Thema',
+    'Worden er persoonsgegevens verwerkt?',
+    'Wettelijke grondslag',
+    'Privacyverklaring',
+    'Tot wanneer is de sensor actief?',
+    'Wilt u nog een sensor melden?',
+]
+
+
+IPROX_FIELDS = IPROX_REGISTRATION_FIELDS + IPROX_PERSON_FIELDS + (IPROX_SENSOR_FIELDS * 5)
+
 
+@dataclasses.dataclass
+class Values:
+    """
+    Wraps a list of values from an excel file (either a row of columns, or
+    column of rows) so that we can provide 'named access' to those values,
+    e.g.
 
-class CsvRowValidationException(Exception):
+    >>> values = Values(['a', 'b', 'c'], [1, 2, 3])
+    >>> values['a']
+    1
+    >>> values['d']
+    KeyError()
+
+    It is possible to have duplicate keys, in which case slicing can be used
+    to get the nth item (0 based) with a particular name, e.g.
+
+    >>> values = Values(['a', 'a', 'a'], [10, 20, 30])
+    >>> values['a', 3]
```

Comment:
Good catch - the example is wrong

Reply:
The reason I wrote the example like this was so that it can be run as a doctest. When adding `` --doctest-modules`` to the pytest invocation it fails 🎉 (first with some other errors in this example 😛, but once I fix those this error is correctly caught).